Userland/Libraries/LibWeb/DOM/AbortSignal.cpp

```diff
@@ -116,4 +116,20 @@ void AbortSignal::follow(JS::NonnullGCPtr<AbortSignal> parent_signal)
     });
 }
 
+// https://dom.spec.whatwg.org/#dom-abortsignal-abort
+WebIDL::ExceptionOr<JS::NonnullGCPtr<AbortSignal>> AbortSignal::abort(JS::VM& vm, JS::Value reason)
+{
+    // 1. Let signal be a new AbortSignal object.
+    auto signal = TRY(construct_impl(*vm.current_realm()));
+
+    // 2. Set signal’s abort reason to reason if it is given; otherwise to a new "AbortError" DOMException.
+    if (reason.is_undefined())
```

Comment:
"if it is given" ...

In other engines, does `abort(undefined)` set the signal to an AbortError, or to `undefined` ?

mattco added a new IDL attribute recently to check for provided/not provided optional arguments that might be needed here if passing undefined has this behavior.

Reply:
I was under the impression that `foo(undefined);` is equivalent to `foo()`?

It looks like we do the right thing given `null` or `undefined` here. Tested with the following script:

```html
<!DOCTYPE html>
<script>
    console.log(`${AbortSignal.abort().reason}`);
    console.log(`${AbortSignal.abort(undefined).reason}`);
    console.log(`${AbortSignal.abort(null).reason}`);
</script>
```

Firefox:
```
AbortError: The operation was aborted. 
AbortError: The operation was aborted.
null
```

Chrome:
```
AbortError: signal is aborted without reason
AbortError: signal is aborted without reason
null
```

Ladybird:
```
[object DOMException]
[object DOMException]
null
```

It would be nice if our DOMException was better formatted, but that's a separate issue.

Reply:
hm. well, sounds like I need to get schooled in some JS details, thanks for testing this!

Reply:
FWIW, according to [this StackOverflow post](https://stackoverflow.com/a/51755964), it looks like it is possible distinguish between an omitted argument and an argument passed as `undefined`.  It doesn't make a difference in this case though.